lib/internal/errors.js

```diff
@@ -828,16 +765,20 @@ function setArrowMessage(err, arrowMessage) {
 // Hide stack lines before the first user code line.
 function hideInternalStackFrames(error) {
   overrideStackTrace.set(error, (error, stackFrames) => {
-    let frames = stackFrames;
+    let result = '';
     if (typeof stackFrames === 'object') {
-      frames = ArrayPrototypeFilter(
-        stackFrames,
-        (frm) => !StringPrototypeStartsWith(frm.getFileName() || '',
-                                            'node:internal'),
-      );
+      for (const frame of stackFrames) {
```

Comment:
for-of is vulnerable to Array.prototype Symbol.iterator modification, this should be a regular for loop, or ArrayPrototypeForEach

Reply:
We do not enforce such usage and I would like to keep them as is.

Reply:
see https://github.com/nodejs/node/pull/46648#discussion_r1122557460